plugin/src/com/microsoft/alm/plugin/external/commands/Command.java

```diff
@@ -240,7 +240,7 @@ public void completed(T result, Throwable error) {
                     throw new ToolEulaNotAcceptedException(error);
                 }
                 if (error instanceof RuntimeException) {
-                    logger.error("Error: {}\nSync stack trace: {}", error, StringUtils.join(Thread.currentThread().getStackTrace(), "\n    at "));
+                    logger.warn("Error: {}\nSync stack trace: {}", error, StringUtils.join(Thread.currentThread().getStackTrace(), "\n    at "));
```

Comment:
why? Looks like an error.

Reply:
It depends on how we handle it in the calling code.

There's two cases when it isn't an error:
- EULA not accepted (caller handles that)
- dollar in path (caller handles that)

I have changed the error level from `error` to `warn` to not produce any unnecessary errors in internal mode. The actual errors are **always** shown (because of an additional `throw` clause below).

Reply:
It is code looks strange we fire warning then exception after that.

Reply:
Are you need log warning there?

Reply:
Look: in the warning message, we're logging the _synchronous stack trace_ from the place the command was called.

The exception from commands usually doesn't contain this information, because the commands are executed on the separate threads. This makes the stack traces from exceptions almost unusable. So, to properly diagnose most command-related errors, we need both
- synchronous stack trace (the warning message)
- exception object thrown

I've tried to change the architecture and include the stack traces separately, but believe me, this is the simplest and most robust way of obtaining them :)

Reply:
ok.